src/olympia/addons/tests/test_tasks.py

```diff
@@ -0,0 +1,72 @@
+import mock
+import os
+import tempfile
+
+from django.conf import settings
+
+from PIL import Image
+
+from olympia.amo.tests import addon_factory, TestCase
+from olympia.amo.tests.test_helpers import get_image_path
+from olympia.addons.tasks import (
+    create_persona_preview_images, save_persona_image)
+
+
+class TestPersonaImageFunctions(TestCase):
+    @mock.patch('olympia.addons.tasks.pngcrush_image')
+    def test_create_persona_preview_image(self, pngcrush_image_mock):
+        addon = addon_factory()
+        addon.modified = self.days_ago(41)
+        # Given an image, a 680x100 and a 32x32 thumbnails need to be generated
+        # and processed with pngcrush.
+        expected_dst1 = tempfile.NamedTemporaryFile(
+            mode='r+w+b', suffix=".png", delete=False, dir=settings.TMP_PATH)
+        expected_dst2 = tempfile.NamedTemporaryFile(
+            mode='r+w+b', suffix=".png", delete=False, dir=settings.TMP_PATH)
+        create_persona_preview_images(
+            src=get_image_path('persona-header.jpg'),
+            full_dst=[expected_dst1.name, expected_dst2.name],
+            set_modified_on=[addon],
+        )
+        # pngcrush_image should not been called twice, once for each
+        # destination thumbnail.
+        assert pngcrush_image_mock.call_count == 2
+        assert pngcrush_image_mock.call_args_list[0][0][0] == (
+            expected_dst1.name)
+        assert pngcrush_image_mock.call_args_list[1][0][0] == (
+            expected_dst2.name)
+
+        assert Image.open(open(expected_dst1.name)).size == (680, 100)
+        assert Image.open(open(expected_dst2.name)).size == (32, 32)
```

Comment:
relying on the python interpreter to close those files, mhmmmmmmm… maybe create a helper to fetch an images dimensions? 

Looking at 

```
[git:master (olympia)] ~/Projects/olympia> acka "Image.open.*?size" src/
src/olympia/devhub/tests/test_views_edit.py
757:        assert Image.open(storage.open(dest)).size == (32, 12)
803:        assert Image.open(storage.open(dest)).size == (48, 48)

src/olympia/devhub/tests/test_tasks.py
96:        assert Image.open(open(dest_image)).size == expected_size

src/olympia/devhub/tasks.py
691:                    'image': Image.open(storage.open(preview.image_path)).size,

src/olympia/devhub/views.py
1043:                actual_size = Image.open(fp).size
```

We have that pattern often enough in our code.